sapi/fpm/fpm/fpm_log.c

```diff
@@ -474,3 +491,39 @@ int fpm_log_write(char *log_format) /* {{{ */
 	return 0;
 }
 /* }}} */
+
+static int fpm_access_log_suppress(struct fpm_scoreboard_proc_s *proc)
+{
+	// Never suppress when query string is passed
+	if (proc->query_string[0] != '\0') {
+		return -1;
+	}
+
+	// Never suppress if request method is not GET or HEAD
+	if (
+		strcmp(proc->request_method, "GET") != 0
+		&& strcmp(proc->request_method, "HEAD") != 0
+	) {
+		return -2;
+	}
+
+	// Never suppress when response code is not 200
+	if (SG(sapi_headers).http_response_code != 200) {
+		return -3;
+	}
```

Comment:
I think it might be better to default to all successful and possibly redirection responses

Reply:
So, >= 400?

Reply:
yeah

Reply:
Sorry for the delay, I guess my gut feel here is that a health check returning 3xx feels likely to be an error and users might expect to see the log entry in that case. Will go with it if you like but feel it's worth stating before I do! In any case I do agree 2xx should be accepted.